crates/cli/src/main.rs

```diff
@@ -528,6 +537,97 @@ async fn start_program(opt: Cli) -> Result<(), AnvilZksyncError> {
             }
         }
     }
+
+    // Use `debug-trace` command to get debug info for a transaction
+    // TODO: clean this up and move to a separate function?
+    if let Command::DebugTrace(args) = command {
+        let params = if args.only_top {
+            json!([args.tx, { "tracer": "callTracer", "tracerConfig": { "onlyTopCall": true } }])
+        } else {
+            json!([args.tx])
+        };
+
+        let rpc_url = args.rpc_url.to_config().url.to_string();
+        let result_value: serde_json::Value =
+            match rpc_call::<serde_json::Value>(&rpc_url, "debug_traceTransaction", params).await {
+                Ok(v) => v,
+                Err(e) => {
+                    sh_println!("debug_traceTransaction failed: {e}");
+                    return Ok(());
+                }
+            };
```

Comment:
I think you can get rid of all custom RPC logic if you just initialize `alloy::Provider` and then call `provider.debug_trace_transaction(...)`. Also it will be a lot more type safe.